pkg/storaged/crypto-keyslots.jsx

```diff
@@ -355,67 +342,67 @@ function edit_tang_adv(client, block, key, url, adv, passphrase) {
-                            <label>
-                                <input type="radio" checked={val === false}
-                                       onChange={event => change(false)} />
-                                { fmt_to_fragments(_("Force remove passphrase in $0"), <b>{key_slot}</b>) }
-                            </label>
-                            <p className="slot-warning" hidden={val !== false}>
-                                Removing a passphrase without confirmation can be dangerous.
-                            </p>
-                        </div>
-                    </div>
-                </div>
-            );
-        }
-    };
-};
+        this.onRemove = this.onRemove.bind(this);
+    }
 
-function remove_passphrase_dialog(block, key) {
-    dialog_open({
-        Title: _("Remove passphrase"),
-        Fields: [
-            RemovePassphraseField("passphrase", key, block_name(block))
-        ],
-        Action: {
-            DangerButton: true,
-            Title: _("Remove"),
-            action: function (vals) {
-                if (vals.passphrase === false)
-                    return slot_remove(block, key.slot);
-                else
-                    return passphrase_remove(block, vals.passphrase);
-            }
-        }
-    });
+    onRemove() {
+        this.setState({ pending: true });
+        const dev = decode_filename(this.props.block.Device);
+        const cmd = ["cryptsetup", "luksKillSlot", dev, this.props.slot.toString()];
+        if (this.state.force)
+            cmd.splice(2, 0, "-q");
+
+        const spawn = cockpit.spawn(cmd, { superuser: true, err: "message" });
+        if (!this.state.force)
+            spawn.input(this.state.passphrase + "\n");
+
+        spawn.then(this.props.onClose)
+                .catch(e => this.setState({ error: e.message }))
+                .always(() => this.setState({ pending: false }));
+    }
+
+    render() {
+        return (
+            <Modal
+                title={cockpit.format(_("Remove passphrase in key slot $0"), this.props.slot)}
+                titleIconVariant="warning"
+                isOpen
+                onClose={this.props.onClose}
+                footer={<>
+                    {
+                        this.state.error && <ModalError dialogError={this.state.error} />
+                    }
+                    <Button key="remove" variant="danger" isLoading={this.state.pending} isDisabled={this.state.pending} onClick={this.onRemove}>{_("Remove")}</Button>
+                    <Button key="cancel" variant="link" isDisabled={this.state.pending} onClick={this.props.onClose}>{_("Cancel")}</Button>
+                </>}
+                position="top" variant="medium"
+            >
+                <Stack hasGutter>
+                    <p>{ fmt_to_fragments(_("Passphrase removal may prevent unlocking $0."), <b>{block_name(this.props.block)}</b>) }</p>
+                    <Checkbox id="run-image-dialog-tty"
```

Comment:
This should be in the Form and then inside a FormGroup.

Reply:
You wil need hasNoPaddingTop property in the FormGroup.

Reply:
> This should be in the Form and then inside a FormGroup.

it already is. It wraps this into `Form` and `FormGroup`. And this was adding additional `FormGroup` into `FormGroup`.

Reply:
Good call with ` hasNoPaddingTop`, with that it looks like this:
![padding](https://user-images.githubusercontent.com/12330670/117284890-9e8f7b80-ae67-11eb-97be-c9f21ec3ac66.png)